File: solution files/Javascript/part-01/task-03-format-output/blueprint_decoder.js
const fs = require("fs");

function decodeBlueprint(filename) {
    const content = fs.readFileSync(filename, "utf8");
    const pattern = /\{\* (.*?) \*\}/g;
    const matches = [...content.matchAll(pattern)];
    const secrets = matches.map((match) => match[1]);
    return secrets;
}

// Enhanced version with error handling
// If file doesn't exist, return empty array and print error message
function decodeBlueprintSafe(filename) {
    try {
        return decodeBlueprint(filename);
    } catch (err) {
        console.log(`Error: File '${filename}' not found.`);
        return [];
    }
}

// Function to format and display secrets in a professional report
// Includes header, separator lines, numbered list, and footer
function displaySecretsReport(secrets) {
    const separator = "=".repeat(50);
    console.log("\n" + separator);
    console.log("🔐 DECODED SECRETS REPORT".padEnd(50));
    console.log(separator);
    console.log(`Total secrets found: ${secrets.length}\n`);

    secrets.forEach((secret, index) => {
        const num = String(index + 1).padStart(2, " ");
        console.log(`  [${num}] ${secret}`);
    });

    console.log("\n" + separator + "\n");
}

// Use it
const secrets = decodeBlueprintSafe("blueprint-data.txt");
displaySecretsReport(secrets);

module.exports = {
    decodeBlueprint,
    decodeBlueprintSafe,
    displaySecretsReport,
};

File: solution files/Python/part-01/task-03-format-output/blueprint_decoder.py
import re


def decode_blueprint(filename):
    with open(filename, "r") as file:
        content = file.read()

    pattern = r"\{\* (.*?) \*\}"
    secrets = re.findall(pattern, content)
    return secrets


# Enhanced version with error handling
# If file doesn't exist, return an empty list and print an error message
def decode_blueprint_safe(filename):
    try:
        return decode_blueprint(filename)
    except FileNotFoundError:
        print(f"Error: File '{filename}' not found.")
        return []


# Function to format and display secrets in a professional report
# Includes header, separator lines, numbered list, and footer
def display_secrets_report(secrets):
    separator = "=" * 50
    print("\n" + separator)
    print("🔐 DECODED SECRETS REPORT".center(50))
    print(separator)
    print(f"Total secrets found: {len(secrets)}\n")

    for i, secret in enumerate(secrets, 1):
        print(f"  [{i:2d}] {secret}")

    print("\n" + separator + "\n")


if __name__ == "__main__":
    secrets = decode_blueprint_safe("blueprint-data.txt")
    display_secrets_report(secrets)
